c10/util/SmallBuffer.h

```diff
@@ -0,0 +1,53 @@
+#pragma once
+#include <type_traits>
+
+/** Helper class for allocating temporary fixed size arrays with SBO.
+ *
+ * This is intentionally much simpler than SmallVector, to improve performace at
+ * the expense of many features:
+ * - No zero-initialization for numeric types
+ * - No resizing after construction
+ * - No copy/move
+ * - No non-trivial types
+ */
+
+namespace c10 {
+
+template <typename T, size_t N>
+class SmallBuffer {
```

Comment:
This is great, and we should use it more in places we are currently using DimVectors.